### DISCUSSION TITLE
Html form only client side

### DISCUSSION THREAD

**Discussion by cwagner22** [remix-run/react-router#12388] [Q&A]
I'm trying to migrate an existing app to react-router 7 framework. The problems is there are lot of forms which I'm not ready to connect to actions. I would like to keep them executing on the client only. But right now when submit is pressed, the page reloads (and I even need to add method="post" to avoid the form data being added as url param).
With remix there was a ClientOnly component which might have worked. (I would have preferred not having to do anything though, like with the Vike plugin)
Any tip?

**Comment by sergiodxa** [comment]
If the route has a clientAction the Form will first execute that, otherwise you can always add an onSubmit and then call `event.preventDefault()`
